src/test/java/edu/ucsb/cs156/example/services/UCSBCurriculumServiceTest.java

```diff
@@ -0,0 +1,80 @@
+package edu.ucsb.cs156.example.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatcher;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.client.RestTemplate;
+
+import edu.ucsb.cs156.example.documents.CourseSearchParams;
+
+import org.springframework.web.client.HttpClientErrorException;
+
+@ExtendWith(SpringExtension.class)
+public class UCSBCurriculumServiceTest {
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @InjectMocks
+    private UCSBCurriculumService ucs;
+
+    class correctHeaders implements ArgumentMatcher<HttpEntity> {

```

Comment:
I like the use of `ArgumentMatcher` here, but can we make this a static class and rename this to follow Java naming conventions? Something like `CorrectHeaderMatcher` would be a better name.

Reply:
Fixed with new commit